dev/io.openliberty.microprofile.metrics.internal.3.0/src/io/openliberty/microprofile/metrics30/internal/helper/BucketManager.java

```diff
@@ -0,0 +1,298 @@
+        // Not running beta edition, throw exception
+        if (!ProductInfo.getBetaEdition()) {
+            if (!issuedBetaWarning) {
+                Tr.warning(tc, "This method is beta and is not available.");
+                issuedBetaWarning = true;
+            }
+        } else {
+            // Running beta exception, issue message if we haven't already issued one for this class
+            if (!issuedBetaMessage) {
+                Tr.info(tc, "BETA: A beta method has been invoked for the class " + this.getClass().getName() + " for the first time.");
+                issuedBetaMessage = true;
+            }
+        }
+
+        return issuedBetaMessage;
+    }
+
+    public void updateTimer(long value) {
+        for (Map.Entry<String, Map<Double, BucketValue>> entry : allBuckets.entrySet()) {
+
+            Map<Double, BucketValue> innerMap = entry.getValue();
+            for (Map.Entry<Double, BucketValue> innerEntry : innerMap.entrySet()) {
+
+                if (innerEntry.getKey() >= (value) / 1000000000.00)
+                    innerEntry.getValue().increment();
+            }
+        }
+    }
+
+    public void updateHistogram(long value) {
+        for (Map.Entry<String, Map<Double, BucketValue>> entry : allBuckets.entrySet()) {
+
+            Map<Double, BucketValue> innerMap = entry.getValue();
+            for (Map.Entry<Double, BucketValue> innerEntry : innerMap.entrySet()) {
+
+                if (innerEntry.getKey() >= (value))
+                    innerEntry.getValue().increment();
+            }
+        }
+    }
+
+    public static class BucketValue {
+        private long value;
+        private final String unit;
+
+        public BucketValue(long value, String unit) {
+            this.value = value;
+            this.unit = unit;
+
+        }
+
+        public double getValue() {
+            return value;
+        }
+
+        public String getUnit() {
+            return unit;
+        }
+
+        public void increment() {
+            value++;
+        }
+    }
+
+    public static class Bucket {
+        private final double value;
+        private final BucketValue bucketValue;
+
+        public Bucket(double value, BucketValue newBucket) {
+            this.value = value;
+            this.bucketValue = newBucket;
+        }
+
+    }
+
+    public Map<String, Map<Double, BucketValue>> getBuckets() {
+        return allBuckets;
+    }
+
+    public Snapshot getBucketsSnap() {
+        return (Snapshot) buckets;
+    }
+
+    /**
+     * Calculates the unit String suffix and conversion factor used for later calculations
+     *
+     * @param unit String that encompasses the unit needed to calculate appropriate conversion factor and value to append
+     * @return Map.Entry<String, Double> that contains the unit string suffix and conversion factor
+     */
+    protected Map.Entry<String, Double> resolveConversionFactorXappendUnitEntry(String unit) {
+
```

Comment:
Can't remember if its' exactly the same, but it's used elsewhere. For GA we can refactor it and place it in a common place to use.